src/tool_cb_hdr.c

```diff
@@ -95,6 +96,49 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
     (void)fflush(heads->stream);
   }
 
+  /*
+   * Write etag to file when --etag-save option is given.
+   */
+  if(etag_save->config->etag_save_file && etag_save->stream) {
+    char *r = NULL;
+
+    /* only match header that starts with expression ETag, Etag or etag */
+    if(strncmp("ETag:", str, 5) == 0 ||
+      strncmp("Etag:", str, 5) == 0 ||
+      strncmp("etag:", str, 5) == 0) {
+
+      /* copy header to my pointer for manipulation */
+      r = strdup(str);
+    }
+
+    /* parse ETag from header, if header line containes an ETag header */
+    if(r) {
+      char *etagc = malloc(strlen(r) + 1);
+
+      if(!etagc) {
+        warnf(
+          etag_save->config->global,
+          "\nFailed allocate memory for etag\n");
+        return 1;
+      }
+
+      /* scan etag header and extract string between double quotes */
+      sscanf(r, "%*512[^\"]\"%512[^\"]\"", etagc);
```

Comment:
... then you could use `memchr()` here (or similar) to find the double quotes.

Reply:
thanks, done